internal/loggerfx/loggerfx.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2023 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package loggerfx
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/goschtalt/goschtalt"
+	"github.com/xmidt-org/sallust"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	deaultConfigPath = "logging"
+)
+
+var (
+	ErrInvalidConfigPath = errors.New("configuration structure path like 'logging' or 'foo.logging' is required")
+)
+
+// Module is function that builds the loggerfx module based on the inputs.  If
+// the configPath is not provided then the default path is used.
+func Module(configPath ...string) fx.Option {
```

Comment:
This is a good pattern, but not something we want to import into every application.  This is the kind of code that belongs in each application that uses these libraries.

The main reason for this is you cannot know a priori how an application is going to organize its modules. An application might put each `http.Server` in its own module, for example.